### PR TITLE
test jinja unsafe setting

### PR DESCRIPTION
**Pull request by chrismeyersfsu**
Add test for all the settings of the jinja2 unsafe filtering.

### CODE REVIEW

File: awx/main/tests/unit/test_tasks.py [hunk 1]
```diff
@@ -309,7 +309,15 @@ class TestExtraVarSanitation(TestJobExecution):
...
 
-    def test_vars_unsafe_by_default(self):
+    @pytest.mark.parametrize('safe_mode,marked_unsafe', [
+        ('always', False),
+        ('template', True),
+        ('never', True)
+    ])
+    def test_vars_unsafe_by_default(self, safe_mode, marked_unsafe):
+        patch = mock.patch('awx.main.tasks.settings.ALLOW_JINJA_IN_EXTRA_VARS', safe_mode)
+        patch.start()
```
Comment (ryanpetrello) [inline awx/main/tests/unit/test_tasks.py:319]: You need to undo this patch afterwards so it doesn't affect other tests, right?

Usually I like to use the context manager so this gets undone automatically i.e.,

```python
with mock.patch(...):
   ...
    self.task.run(self.pk)
```

**Comment by AlanCoding** [comment]
recheck

File: awx/main/tests/unit/test_tasks.py (hunk 1 above)
Comment (AlanCoding) [reply awx/main/tests/unit/test_tasks.py:319]: yeah, that sounds safer to me and very easy to do as a context manager

**Comment by ryanpetrello** [comment]
Closing this @chrismeyersfsu, reopen if you find time to give it another shot.

**Pull request closed without merging**